File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        res = 0
        left = 0
        right = len(height) - 1
        while left < right:
            cur_h = min(height[left], height[right])
            cur_l = right - left
            cur_storage = cur_h * cur_l
            res = max(res, cur_storage)
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return res

File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    const calWater = (left, right, distant) => {
        const h = Math.min(right, left);
        return h * distant
    }
    let l= 0;
    let r = height.length - 1;
    let res = 0;
    while (l < r ) {
        const temp = calWater(height[l], height[r], r-l)
        res = Math.max(res, temp)
        if (height[l] <height[r]) {
            l ++
        } else {
            r --
        }
    }
    return res

};